flink-core/src/main/java/org/apache/flink/types/parser/IntParser.java

```diff
@@ -38,19 +38,28 @@ public int parseField(byte[] bytes, int startPos, int limit, byte[] delimiter, I
 
 		final int delimLimit = limit-delimiter.length+1;
 
+		if (bytes.length == 0) {
```

Comment:
I see. This is probably because an empty test string causes the test to call the parser with an 0-length array.
We could add a dedicated `testEmptyField` test method to the `ParserTestBase` and remove the empty Strings from the set of invalid inputs.